status-handler.js

```diff
@@ -143,6 +151,103 @@ class StatusHandler {
     this.log.info(`Ignored status event, no build occurred`, data);
   }
 
+  /**
+   * Check if there are third-party apps that
+   * subscribed for webhooks for a specific package
+   *
+   * @function _shouldSendWebhook
+   * @param {String} pkg - The name of the package
+   * @returns {Boolean} the computed value
+   */
+  _shouldSendWebhook(pkg) {
+    const webhooks = this.webhooks[pkg];
+    if (!webhooks || webhooks.length === 0) {
+      this.log.info(`No webhooks for pkg ${pkg}`);
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Check if a build is in a queued status
+   *
+   * @function _isBuildQueued
+   * @param {Object} build - Build params
+   * @param {String} build.pkg - The package name
+   * @param {String} build.version - The package version
+   * @param {String} build.env - The build environnement
+   * @returns {String} the package name
+   */
+  async _isBuildQueued(build) {
+    // Fetch list of previous events for a certain build
+    const previousEvents = await this.models.StatusEvent.findAll(build);
+
+    // Ensure that last event is the 'queued' event from carpentd
+    const lastEvent = previousEvents[previousEvents.length - 1];
+    if (lastEvent && lastEvent.message === 'Builds Queued') return true;
```

Comment:
we need to also check the event that triggered this function call to ensure its the first event that comes from `carpenterd-worker`, i believe its fetch tarball. The reason is that `Builds Queued` could be the last message created but something happens where the builds werent picked up off the queue or they were dropped. In this case we might have to dist-tag again in order to make it work. With the current logic, the first message that carpenterd would send when it started the build process again would trigger the webhook incorrectly

Reply:
also, since we do `StatusEvent.create` before we dispatch the webhook, due to this, we may have the first eventt from carpenterd-worker as the `lastEvent` rather than the one we are looking for here, the carpenterd' Builds Queued. We should account for this. Maybe inspect the last 2 events and make it more robust if theres some timing stuff with the database @JacopoDaeli

Reply:
👍 What this event would look like @jcrugzz ?